src/material/slider/slider.ts

```diff
@@ -140,7 +140,7 @@ export class MatSlider implements AfterViewInit, OnDestroy, _MatSlider {
     return this._min;
   }
   set min(v: number) {
-    const min = isNaN(v) ? this._min : v;
+    const min = !v || isNaN(v) ? this._min : v;
```

Comment:
`!v` will be true for `0`, which won't trigger the change of `this._min`.
This can be a problem if `min` was previously set  to `-1` and then trying to set it to `0`.

Consider checking `v` against `undefined` and `null` values.

```suggestion
    const min = v === undefined || v === null || isNaN(v) ? this._min : v;
```

Reply:
Agreed, I first thought of this whether this change is required or not. But yeah I'll make the suggested changes.